src/components/chats/ChatRoom/RepliedMessage.tsx

```diff
@@ -0,0 +1,64 @@
+import Button from '@/components/Button'
+import { getPostQuery } from '@/services/api/query'
+import { generateRandomColor } from '@/utils/random-colors'
+import { generateRandomName } from '@/utils/random-name'
+import { ComponentProps } from 'react'
+import { BsFillReplyFill } from 'react-icons/bs'
+import { HiXMark } from 'react-icons/hi2'
+import { getChatItemId, scrollToChatItem } from '../helpers'
+
+export type RepliedMessageProps = ComponentProps<'div'> & {
+  replyChatId: string
+  closeReply: () => void
+  scrollContainer?: React.RefObject<HTMLElement | null>
+}
+
+export default function RepliedMessage({
+  replyChatId,
+  closeReply,
+  scrollContainer,
+}: RepliedMessageProps) {
+  const { data } = getPostQuery.useQuery(replyChatId)
+  const chatContent = data?.content?.body
+  const chatSenderAddr = data?.struct.ownerId
+  const senderColor = generateRandomColor(chatSenderAddr ?? '')
+  const name = generateRandomName(chatSenderAddr ?? '')
```

Comment:
Updated in [877e68d](https://github.com/dappforce/grillchat/pull/38/commits/877e68d09abd5406b233101ec594e4efd345ff71)

for the others, like decodeSecretKey, I don't really want to allow sending null because if so, it will seem like this `decodeSecretKey` can be done with null `secretKey`, which is weird. I think its better most of the time for the outside code to validate that its really a string before calling the function rather than normalizing it.

well, in my case, I got a bit lazy and just put `?? ''` 😅, but in the crucial parts it should be checked before, so I think its better to keep it just string in the param.